openedx/core/djangoapps/catalog/utils.py

```diff
@@ -101,18 +111,74 @@ def get_program_types(user=None):  # pylint: disable=redefined-builtin
         return []
 
 
-def get_programs_data(user=None):
-    """Return the list of Programs after adding the ProgramType Logo Image"""
+def _get_program_instructors(program):
+    """
+    Returns the list of instructor from cached if cache key exists otherwise
+    iterate over the courses and return all the instructors of each course run
+    """
+    cache_key = 'program.instructors.{program_id}'.format(
+        program_id=program.get('uuid')
+    )
+
+    program_instructors_dict = {}
+    program_instructors_list = cache.get(cache_key, [])
+    if program_instructors_list:
+        return program_instructors_list
+
+    module_store = modulestore()
+    for course_run_key in _get_all_course_run_keys(program):
+        course_descriptor = module_store.get_course(course_run_key)
+        if course_descriptor:
+            course_instructors = getattr(course_descriptor, 'instructor_info', {})
+            # Deduplicate program instructors using instructor name
+            program_instructors_dict.update(
+                {instructor.get('name'): instructor for instructor in course_instructors.get('instructors', [])}
+            )
+    program_instructors_list = program_instructors_dict.values()
+    cache.set(cache_key, program_instructors_list)
+    return program_instructors_list
+
+
+def _get_all_course_run_keys(program):
+    """
+    Returns the course keys of all the course runs of a program.
+    """
+    keys = []
+    for course in program.get('courses', []):  # pylint: disable=E1101
+        for course_run in course.get('course_runs', []):
+            keys.append(CourseKey.from_string(course_run.get('key')))
+    return keys
+
+
+def get_program_details(user=None, program_id=None):
+    """
+    This will return the program details with its corresponding program type and instructors
+    """
+    program = get_programs(user, program_id, status="active")
+    if not program:
+        return None
 
-    programs_list = get_programs(user)
-    program_types = get_program_types(user)
+    program['type'] = next(
+        program_type
+        for program_type in get_program_types(user)
+        if program_type['name'] == program['type']
+    )
+    program['instructors'] = _get_program_instructors(program)
+    return program
 
-    program_types_lookup_dict = {program_type["name"]: program_type for program_type in program_types}
 
-    for program in programs_list:
-        program["logo_image"] = program_types_lookup_dict[program["type"]]["logo_image"]
+def get_active_programs_list(user=None):
+    """
+    Return the list of active Programs after adding the ProgramType Logo Image
+    """
+    programs = get_programs(user, status="active")
+    if not programs:
+        return []
 
-    return programs_list
+    program_types = {program_type["name"]: program_type for program_type in get_program_types(user)}
+    for program in programs:
+        program["logo_image"] = program_types[program["type"]]["logo_image"]
```

Comment:
Instead of setting the 'logo_image' key, set the 'type' key to the full program type dict.